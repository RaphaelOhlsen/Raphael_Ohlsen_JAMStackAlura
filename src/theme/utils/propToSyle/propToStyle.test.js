import propToStyle from './index';

describe('propToStyle()', () => {
  describe('when recives a simple argument', () => {
    test('and it is a string', () => {
      const propToSyleResult = propToStyle('textAlign');
      // <Text textAlign="center" />
      const componentProps = { textAlign: 'center' };
      const styleResult = propToSyleResult(componentProps);
      expect(styleResult).toEqual({ textAlign: 'center' });
    });
    test('and it is a number', () => {
      const propToSyleResult = propToStyle('flex');
      // <Box flex={1} />
      const componentProps = { flex: 1 };
      const styleResult = propToSyleResult(componentProps);
      expect(styleResult).toEqual({ flex: 1 });
    });
  });
  describe('when recives an argument with breakpoints', () => {
    test('with one breakpoint resolution', () => {
      const propToStypeResult = propToStyle('textAlign');
      // <Box margin={xs: 10px} />
      const componentProps = { textAlign: { xs: 'center' } };
      const styleResult = propToStypeResult(componentProps);
      expect(styleResult).toMatchSnapshot();
    });
  });
  describe('when recives an argument with breakpoints', () => {
    test('with two or more breakpoint resolution', () => {
      const propToStypeResult = propToStyle('textAlign');
      // <Box textAlign={xs: 'center', md: 'right'} />
      const componentProps = { textAlign: { xs: 'center', md: 'right' } };
      const styleResult = propToStypeResult(componentProps);
      expect(styleResult).toMatchSnapshot();
    });
  });
});
