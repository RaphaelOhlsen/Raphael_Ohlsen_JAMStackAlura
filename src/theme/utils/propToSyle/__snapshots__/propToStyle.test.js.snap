// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`propToStyle() when recives an argument with breakpoints with one breakpoint resolution 1`] = `
Array [
  "@media screen and (min-width:",
  "0",
  "px){",
  "text-align: center;",
  "}",
]
`;

exports[`propToStyle() when recives an argument with breakpoints with two or more breakpoint resolution 1`] = `
Array [
  "@media screen and (min-width:",
  "0",
  "px){",
  "text-align: center;",
  "}",
  "@media screen and (min-width:",
  "768",
  "px){",
  "text-align: right;",
  "}",
]
`;
